### PR TITLE
Fix/search widget

### PR DESCRIPTION
**Pull request by microooji**
@GalGend 
ui bug fixes:
1. Chevron is not clickable, but should be
2. The address-search placeholder text is wrong. (should be “לדוגמא: הירקון 16, חיפה”)
3. Purple should be darker
4. font size + chevron of search mode should be bigger
5. Width is shorter that design
6. Mode-change button is not aligned with the placeholder text
7. No tooltip error text when submitting a bad block number. Expected:
8. When submitting the form with an invalid parcel number, the red error border is around the block number instead of the parcel

### CODE REVIEW

**Comment by GalGend** [comment]
Hi @microooji , does this work for you? When I run the block/ parcel search I get this error at BE, let me know if it works for you, as I had this error before and resolved it, so not sure where is the problem:

```s:526:10)\n    at Socket.emit (events.js:315:20)\n    at addChunk (_stream_readable.js:309:12)\n    at readableAddChunk (_stream_readable.js:284:9)\n    at Socket.Readable.push (_stream_readable.js:223:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:188:23)\n    --------------------\n    at Protocol._enqueue (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/mysql/lib/protocol/Protocol.js:144:48)\n    at Connection.query (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/mysql/lib/Connection.js:198:25)\n    at /Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/knex/lib/dialects/mysql/index.js:134:18\n    at new Promise (<anonymous>)\n    at Client_MySQL._query (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/knex/lib/dialects/mysql/index.js:128:12)\n    at Client_MySQL.query (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/knex/lib/client.js:168:17)\n    at Runner.query (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/knex/lib/runner.js:151:36)\n    at /Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/knex/lib/runner.js:40:23\n    at /Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/knex/lib/runner.js:277:24"],"level":"error","service":"user-service"}
{"message":["General error","ER_GIS_DIFFERENT_SRIDS","select `id`, `PLAN_COUNTY_NAME`, `PL_NUMBER`, `PL_NAME`, `PLAN_CHARACTOR_NAME`, `geom`, `plan_display_name`, `goals_from_mavat`, `main_details_from_mavat`, `status`, `updated_at`, `data`, ST_Distance(geom, ST_GeomFromText(\"POINT (35.289849820007504 32.70422667398278)\",4326))*111195 as distance from `plan` where `geo_search_filter` in (false) order by `distance` asc limit 20 - ER_GIS_DIFFERENT_SRIDS: Binary geometry function st_distance given two geometries of different srids: 0 and 4326, which should have been identical.","::ffff:127.0.0.1 - GET:/plan/ {\"page\":\"1\",\"distancePoint\":\"35.289849820007504,32.70422667398278\"}","Error: ER_GIS_DIFFERENT_SRIDS: Binary geometry function st_distance given two geometries of different srids: 0 and 4326, which should have been identical.\n    at Query.Sequence._packetToError (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)\n    at Query.ErrorPacket (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)\n    at Protocol._parsePacket (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/mysql/lib/protocol/Protocol.js:291:23)\n    at Parser._parsePacket (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/mysql/lib/protocol/Parser.js:433:10)\n    at Parser.write (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/mysql/lib/protocol/Parser.js:43:10)\n    at Protocol.write (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/mysql/lib/protocol/Protocol.js:38:16)\n    at Socket.<anonymous> (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/mysql/lib/Connection.js:88:28)\n    at Socket.<anonymous> (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/mysql/lib/Connection.js:526:10)\n    at Socket.emit (events.js:315:20)\n    at addChunk (_stream_readable.js:309:12)\n    at readableAddChunk (_stream_readable.js:284:9)\n    at Socket.Readable.push (_stream_readable.js:223:10)\n    at TCP.onStreamRead (internal/stream_base_commons.js:188:23)\n    --------------------\n    at Protocol._enqueue (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/mysql/lib/protocol/Protocol.js:144:48)\n    at Connection.query (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/mysql/lib/Connection.js:198:25)\n    at /Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/knex/lib/dialects/mysql/index.js:134:18\n    at new Promise (<anonymous>)\n    at Client_MySQL._query (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/knex/lib/dialects/mysql/index.js:128:12)\n    at Client_MySQL.query (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/knex/lib/client.js:168:17)\n    at Runner.query (/Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/knex/lib/runner.js:151:36)\n    at /Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/knex/lib/runner.js:40:23\n    at /Users/admin/Documents/Meirim/Workspace/meirim/server/node_modules/knex/lib/runner.js:277:24"],"level":"error","service":"user-service"}
```

**Comment by microooji** [comment]
Hi @GalGend ! When I developing FE, I don’t run a local BE, I don’t have a base dump and environment files

**Comment by microooji** [comment]
@GalGend I pushed new features:
1. Add new search mode
2. Disappear with scrolling down, appears when scrolling up
3. Make sure it’s looking good in mobile
4. New loader/spinner when fetching plans

File: client/src/scenes/Plans.jsx [hunk 1]
```diff
@@ -149,14 +358,45 @@ class Plans extends Component {
...
+        this.lastScrollY = window.scrollY;
         // init location service
         locationAutocompleteApi.init();
 
         // if there is no valid query string params to search by run default search
         if (!this.loadQsSearchParams()) {
             this.loadPlans(this.state.pageNumber);
         }
+
+        document.addEventListener('scroll', this.handleScrollEvent);
```
Comment (GalGend) [inline client/src/scenes/Plans.jsx:395]: Maybe the're a react hook for that?

File: client/src/scenes/Plans.jsx [hunk 2]
```diff
@@ -149,14 +358,45 @@ class Plans extends Component {
         }
     }
 
+    handleScrollEvent = (e) => {
```
Comment (GalGend) [inline client/src/scenes/Plans.jsx:361]: Why is this needed?

File: client/src/pages/Homepage/SearchBox.jsx [hunk 3]
```diff
@@ -224,249 +301,325 @@ const InputWrapper = styled.div`
...
+        }
+    }
+
+    useEffect(() => {
+        locationAutocompleteApi.init();
+    }, []);
+
+    const [dropdownIsOpen, setDropdownIsOpen] = useState(false);
+
+    return (
```
Comment (GalGend) [inline client/src/pages/Homepage/SearchBox.jsx:481]: Why on clicking search (block/ parcel or address ) the page is refreshed? The search should be performed without a refresh

File: client/src/pages/Homepage/TopSection.jsx [hunk 4]
```diff
@@ -132,12 +132,12 @@ const TopSection = () => {
 			<Cover>
                 <H3>{t.homepageMainTopTitle}</H3>
 				<H1>{t.homepageMainTitle}</H1>
-				<SearchBox />
+				<SearchBox backgroundColor="#4D20B2" wrapperPadding="20px" />
```
Comment (GalGend) [inline client/src/pages/Homepage/TopSection.jsx:135]: You can add the stylying to the SearchBox component itself

**Review by GalGend — CHANGES_REQUESTED** [review]
Hi Roman this looks great. Added a few comments.

The search on plans page should be preformed without a page refresh.

Thank you

File: client/src/scenes/Plans.jsx (hunk 1 above)
Comment (microooji) [reply client/src/scenes/Plans.jsx:395]: @GalGend we cant use react hooks in the class components. componentDidMount it is like useEffect. if you want, I can rewrite PlansHeader as a functional component.

File: client/src/pages/Homepage/SearchBox.jsx (hunk 3 above)
Comment (microooji) [reply client/src/pages/Homepage/SearchBox.jsx:483]: @GalGend fixed, someone used window.location.href.

File: client/src/pages/Homepage/TopSection.jsx (hunk 4 above)
Comment (microooji) [reply client/src/pages/Homepage/TopSection.jsx:135]: @GalGend we need to use props because this component is used in 3 different places and they have different background color and padding

File: client/src/scenes/Plans.jsx (hunk 2 above)
Comment (microooji) [reply client/src/scenes/Plans.jsx]: I removed (e)
